units/src/amount/signed.rs

```diff
@@ -337,6 +336,16 @@ impl SignedAmount {
             Ok(Amount::from_sat(self.to_sat() as u64))
         }
     }
+
+    /// Checks if the amount is above the minimum and below the maximum value.
+    /// Returns `None` if it is outside the allowed range.
+    const fn check_min_max(self) -> Option<SignedAmount> {
```

Comment:
I think a more consistent name is `from_sats_checked` like I added here: https://github.com/rust-bitcoin/rust-bitcoin/pull/3703

Reply:
I think `from_sats_checked` is related to addressing #3696.  This is just a private function created so the code is not repeated for the min/max check in the various other functions.  But if the public function is created it could be used for the check as well I guess.

Reply:
But if we add a check for MIN to `from_sats_checked`, then we don't need `check_min_max`.

Reply:
As a private helper function this looks fine to me. Since its private I'd shorten the docs though. Also checking min first seems a bit cleaner to me

```rust
    /// Checks the amount is within the allowed range.
    const fn check_min_max(self) -> Option<SignedAmount> {
        if self.0 < Self::MIN.0 || self.0 > Self::MAX.0 {
            None
        } else {
            Some(self)
        }
    }
```